modules/cloudsql-instance/variables.tf

```diff
@@ -13,6 +13,15 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+variable "activation_policy" {
+  description = "(Optional) This specifies when the instance should be active. Can be either ALWAYS, NEVER or ON_DEMAND. Default is ALWAYS"
+  default  = "ALWAYS"
+  validation {
+    condition     = var.activation_policy == "NEVER" || var.activation_policy == "ON_DEMAND" || var.activation_policy == "ALWAYS" 
+    error_message = "activation_policy must be ALWAYS, NEVER or ON_DEMAND"
```

Comment:
I'd keep null as the default.

Error message must end in period and the first letter must be uppercase.

Reply:
on GCP provider side that null is defaulted to ALWAYS. I Defaulted to ALWAYS for this reason (transparency). you cannot use a null value on provider, I'lll correct the error message